lib/matplotlib/colorbar.py

```diff
@@ -440,9 +440,10 @@ def __init__(self, ax, mappable=None, *, cmap=None,
             linewidths=[0.5 * mpl.rcParams['axes.linewidth']])
         self.ax.add_collection(self.dividers)
 
-        self.locator = None
-        self.minorlocator = None
-        self.formatter = None
+        self._locator = None
+        self._minorlocator = None
+        self._formatter = None
+        self._minorformatter = None
```

Comment:
I haven't traced minor formatter all the way through the piping yet, however, I added it to be parallel with the other properties.  I get a little nervous about how often the colorbar changes state, particularly now that it seems to change state if the mappable's norm is changed ;-)  If I change the norm on the mappable, but I have manually set locator, should that stay or should we overwrite it?  I think now it stays unless self._locator is None, in which case we use the default  appropriate to the norm.  We could simplify our lives by saying that any norm changes invalidates any manual setting of the colorbar locators/formatters, and that the user would have to redo those.  I think that is what happens if a user changes the scale of an axis, so it would be consistent here.

Reply:
I agree that all of this is quite confusing with when things are validated/invalidated in the Colorbar pipeline. What I was trying to get at here was only for the minorformatter. I think you should also set it to None down below too where the other locators/formatters are also reset, since you are now keeping track of it.